File: WebApp/src/app/cadastros/models/produto.ts
import { Marca } from "./marca";
import { Categoria } from "./categoria";
import { Estoque } from "./estoque";
import { TabelaPreco } from "./tabelaPreco";

export class Produto {
    cdProduto: number;
    dsProduto: string;
    dsModelo: string;
    dsInfAdicionais: string;
    fgAtivo: boolean;
    fgTipo: string;
    fgBicicleta: boolean;
    vlServico: number;
    cdMarca: number;
    marca: Marca;
    cdCategoria: number;
    categoria: Categoria;
    dtRegistro: Date;
    dtAlteracao: Date;
    estoqueList: Array<Estoque>;
    tabelaPrecoList: Array<TabelaPreco>;
}

File: BikeManagerAPI/BikeManagerAPI/Models/Produto.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace BikeManagerAPI.Models
{
    public partial class Produto
    {
        public Produto()
        {
            Bicicleta = new HashSet<Bicicleta>();
            ItemNotaEntrada = new HashSet<ItemNotaEntrada>();
            ItemOrcamento = new HashSet<ItemOrcamento>();
            ItemOrdemServico = new HashSet<ItemOrdemServico>();
            TabelaPrecoList = new HashSet<TabelaPreco>();
        }

        public int CdProduto { get; set; }
        public int? CdMarca { get; set; }
        public int? CdCategoria { get; set; }
        public string DsProduto { get; set; }
        public string DsModelo { get; set; }
        public string DsInfAdicionais { get; set; }
        public bool? FgAtivo { get; set; }
        public bool FgBicicleta { get; set; }
        public string FgTipo { get; set; }
        public decimal? VlServico { get; set; }
        public DateTime DtRegistro { get; set; }
        public DateTime? DtAlteracao { get; set; }
        public Categoria Categoria { get; set; }
        public Marca Marca { get; set; }
        [JsonIgnore]
        public ICollection<Bicicleta> Bicicleta { get; set; }
        [JsonIgnore]
        public ICollection<ItemNotaEntrada> ItemNotaEntrada { get; set; }
        [JsonIgnore]
        public ICollection<ItemOrcamento> ItemOrcamento { get; set; }
        [JsonIgnore]
        public ICollection<ItemOrdemServico> ItemOrdemServico { get; set; }
        [JsonIgnore]
        public ICollection<TabelaPreco> TabelaPrecoList { get; set; }
    }
}
